dynamic_programming/knapsack.c

```diff
@@ -0,0 +1,227 @@
+                K[i][w] = K[i - 1][w];
+            }
+        }
+    }
+
+    print_knapsack_result(maxWeight, itemCount, items, K);
+    return K[itemCount][maxWeight];
+}
+
+/**
+ * @brief Print 0-1 Knapsack algorithm result
+ * @param maxWeight the capacity of the knapsack
+ * @param itemCount the number of items(Item structure)
+ * @param items an array of Item structure
+ * @param K the 2D array storing the optimal solution
+ * @returns void
+ */
+void print_knapsack_result(int maxWeight, int itemCount, struct Item items[],int K[][MAXCAPACITY + 1]){
+    printf("Knapsack execution result:\n");
+    for (int i = 0; i < itemCount + 1; i++)
+    {
+        for (int j = 0; j < maxWeight + 1; j++)
+        {
+            printf("%2d ", K[i][j]);
+        }
+        printf("\n");
+    }
+    printf("\n");
+
+    // Print the maximum value of the knapsack
+    printf("Maximum Value: %d\n", K[itemCount][maxWeight]);
+    // Print the combination of items in the knapsack
+    print_knapsack_item(maxWeight, items, itemCount, K);
+}
+
+/**
+ * @brief Print input items
+ * @param items an array of Item structure
+ * @param itemCount the number of items(Item structure)
+ * @returns void
+ */
+void print_input_struct_Items(struct Item items[], int itemCount)
+{
+    printf("Input Item Structures:\n");
+    for (int i = 0; i < itemCount; i++)
+    {
+        printf("Item %d (Weight: %dkg, Value: %d)\n", i + 1, items[i].weight,
+               items[i].value);
+    }
+    printf("\n");
+}
+
+/**
+ * @brief Print the combination of items in the knapsack
+ * @param maxWeight the capacity of the knapsack
+ * @param items an array of Item structure
+ * @param itemCount the number of items(Item structure)
+ * @param K the 2D array storing the optimal solution
+ * @returns void
+ */
+void print_knapsack_item(int maxWeight, struct Item items[], int itemCount,
+                         int K[][MAXCAPACITY + 1])
+{
+    // Backtracking from the final maximum value to track the value changes
+    int i = itemCount;  // Number of items (maximum to 0)
+    int w = maxWeight;  // Temporary capacity of the knapsack (maximum to 0)
+    printf("Items in the knapsack(%dkg):\n", maxWeight);
+    while (i > 0 && w > 0)
+    {
+        // Check if the i-th item is selected
+        if (K[i][w] != K[i - 1][w])
+        {
+            /* If K[i][j] != K[i-1][j], it means the i-th item is included at
+             * the previous (w-items[i-1].weight) weight. */
+            printf("Item %d (Weight: %dkg, Value: %d)\n", i, items[i - 1].weight,
+                   items[i - 1].value);
+            w = w - items[i - 1].weight;  // Move to the weight excluding the
+                                          // i-th item from now on
+        }
+        /* If K[i][w] == K[i-1][maxWeight], it means the i-th item is not
+         * included in the knapsack: w remains the same. */
+        i--;
+    }
+    printf("\n");
+}
+
+/**
+ * @brief Self-tests implementations
```

Comment:
A bit of a nitpick. 😅
```suggestion
 * @brief Self-test implementations
```

Reply:
Thank you 😊 Fixed it !